spec/services/purchase_order_creator_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe 'PurchaseOrderCreator' do
+  describe '#call' do
+    let(:cart) { create(:cart) }
+    let(:user) { cart.user }
+    let(:address) { nil }
+
+    it 'creates order' do
+      expect do
+        PurchaseOrderCreator.call({ user: user, cart: cart, address: address })
+      end.to change { Order.count }
```

Comment:
É importante verificar se foi criado a order com os valores corretos, sem isso o seu teste não cobre exatamente o que precisa. Esse teste precisa pegar erros nas linhas https://github.com/feperessim/refactoring-exercise-for-trainees/pull/1/files#diff-3984a42cd9a6b46b1bba508f0a597a5022ed465b8ee51b3087112fc72a376732R16-R24 por exemplo.